v3/src/components/tool-shelf/tiles-list-button.tsx

```diff
@@ -0,0 +1,107 @@
+    const tileLayout = tileRow?.getTileLayout(tileId)
+    isFreeTileLayout(tileLayout) && tileLayout.setMinimized(false)
+  }
+
+  // const getTitle = (tile: ITileModel) => {
+  //   // return tile.title ?? tile.content?.name ?? tile.content?.data?.title
+
+  //   const { title } = tile || {}
+  //   // if (tile.title) return tile.title
+  //   // else if (name in tile.content) return tile.content.name
+  //   // else if (title in tile.content.data) return tile.content.data.title
+  //   const tileName = () => {
+  //     let model
+  //     switch (tile.content.type) {
+  //       case kSliderTileType:
+  //         model = isSliderModel(tile.content) ? tile.content : undefined
+  //         return model?.name || t("DG.DocumentController.sliderTitle")
+  //       case kCalculatorTileType:
+  //         model = isCalculatorModel(tile.content) ? tile.content : undefined
+  //         return model?.name || t("DG.DocumentController.calculatorTitle")
+  //       case kCaseTableTileType:
+  //       case kCaseCardTileType:
+  //         model = isCaseTableModel(tile.content) ? tile.content : undefined
+  //         return model?.data?.title || t("DG.DocumentController.caseTableTitle")
+  //       case kGraphTileType:
+  //         model = isGraphContentModel(tile.content) ? tile.content : undefined
+  //         return tile.title || t("DG.DocumentController.graphTitle")
+  //       case kMapTileType:
+  //         model = isMapContentModel(tile.content) ? tile.content : undefined
+  //         return tile.title || t("DG.DocumentController.mapTitle")
+  //     }
+  //   }
+  //   return title || tileName() || t("DG.WebView.defaultTitle")
+  //   // return title || name || dataTitle || "no title"
+  // }
+
+  return (
+    <>
+      <Menu isLazy>
+        <MenuButton
+          className="tool-shelf-button tiles-list-menu"
+          title={t("DG.ToolButtonData.tileListMenu.toolTip")}
+          data-testid="tool-shelf-button-tiles"
+        >
+          <OptionsIcon />
+          <ToolShelfButtonTag
+            bg={kRightButtonBackground}
+            className="tiles-list"
+            label={t("DG.ToolButtonData.tileListMenu.title")}
+          />
+        </MenuButton>
+        <MenuList data-testid="tiles-list-menu">
+          {tilesArr?.map((tile) => {
+            const tileType = tile?.content.type
+            const Icon = getTileComponentIcon(tileType)
+            const tileInfo = getTileContentInfo(tileType)
+            console.log("tileInfo", tileInfo)
+            const title = tileInfo?.getTitle(tile)
+            return (
+              <MenuItem key={tile?.id} data-testid="tiles-list-menu-item"
+                  onClick={()=>handleSelectTile(tile.id)}>
+                {(Icon && <Icon className={`tile-list-menu-icon ${tile.content.type}`}
+                              data-testid="tile-list-menu-icon"/>) ||
+                            <WebViewIcon className="tile-list-menu-icon WebView" data-testid="tile-list-menu-icon"/>}
```

Comment:
```suggestion
                <Icon className={`tile-list-menu-icon ${iconClass}`} data-testid="tile-list-menu-icon"/>
```
Needs to be combined with the previous suggestion.